packages/webpack-plugin/lib/runtime/components/react/mpx-portal/portal-host.tsx

```diff
@@ -0,0 +1,125 @@
+import { useEffect, useRef, ReactNode } from 'react'
+import {
+  View,
+  DeviceEventEmitter,
+  EventSubscription,
+  NativeEventEmitter,
+  StyleSheet
+} from 'react-native'
+import PortalManager from './portal-manager'
+import { useNavigation } from '@react-navigation/native'
+import { PortalManagerContextValue, PortalContext } from '../context'
+
+export type PortalHostProps = {
+  children: ReactNode
+}
+
+type addIdsMapsType = {
+  [key: number]: number[]
+}
+
+export type Operation =
+  | { type: 'mount'; key: number; children: ReactNode }
+  | { type: 'update'; key: number; children: ReactNode }
+  | { type: 'unmount'; key: number }
+
+// events
+const addType = 'MPX_RN_ADD_PORTAL'
+const removeType = 'MPX_RN_REMOVE_PORTAL'
+const updateType = 'MPX_RN_UPDATE_PORTAL'
+// fix react native web does not support DeviceEventEmitter
+const TopViewEventEmitter = DeviceEventEmitter || new NativeEventEmitter()
+
+const styles = StyleSheet.create({
+  container: {
+    flex: 1
+  }
+})
+
+class PortalGuard {
+  private nextKey = 10000
+  add = (e: ReactNode) => {
+    const key = this.nextKey++
```

Comment:
if(this.listenerCount === 0) {
this.queue.push({type:addType, key, payload:e})
}